### PR TITLE
news api model remote

### PR DESCRIPTION
**Pull request by RadwaMohamed29**

### CODE REVIEW

File: app/src/main/java/com/example/news/model/CachedData.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.example.news.model
+
+import androidx.room.PrimaryKey
+
+data class CachedData(
```
Comment (RemonShehata) [inline app/src/main/java/com/example/news/model/CachedData.kt:8]: please create a separate model for the database

**Review by RemonShehata — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/news/model/NewsApi.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package com.example.news.model
+
+import androidx.room.Entity
+
+@Entity(tableName = "News")
```
Comment (AbdulrahmanAli19) [inline app/src/main/java/com/example/news/model/NewsApi.kt]: This annotation should not be here

**Review by AbdulrahmanAli19 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/news/model/NewsApi.kt (hunk 2 above)
Comment (RadwaMohamed29) [reply app/src/main/java/com/example/news/model/NewsApi.kt]: fixed

File: app/src/main/java/com/example/news/model/CachedData.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+
+
+@Entity(tableName = "News")
+data class CachedData(
+    @PrimaryKey(autoGenerate = true)
+    val id:Int,
+    val cachedData: List<Article>
```
Comment (RemonShehata) [inline app/src/main/java/com/example/news/model/CachedData.kt]: create a new article class for the database entity. @RadwaMohamed29

File: app/src/main/java/com/example/news/model/CachedData.kt (hunk 1 above)
Comment (RadwaMohamed29) [reply app/src/main/java/com/example/news/model/CachedData.kt:8]: fixed

File: app/src/main/java/com/example/news/model/CachedData.kt (hunk 3 above)
Comment (RadwaMohamed29) [reply app/src/main/java/com/example/news/model/CachedData.kt]: fixed

File: app/src/main/java/com/example/news/data/local/Converter.kt [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+//    fun articleListToGson(value: List<Article>) = Gson().toJson(value)
+//
+//    @TypeConverter
+//    fun gsonToArticleList(value: String) = Gson().fromJson(value, Array<Article>::class.java).toList()
+//
+//    //@TypeConverter
+//    //fun convertApiTo
+//}
+
+fun fromNewsToEntity(newsApi: NewsApi):CachedData{
```
Comment (RemonShehata) [inline app/src/main/java/com/example/news/data/local/Converter.kt]: did you copy this from abelrahaman sayed? 😄

Comment (RemonShehata) [reply app/src/main/java/com/example/news/data/local/Converter.kt]: see comments here https://github.com/AhmedAbdallah125/7asryan/pull/8#pullrequestreview-921865661

File: app/src/main/java/com/example/news/model/CachedData.kt [hunk 5]
```diff
@@ -0,0 +1,48 @@
+package com.example.news.model
+
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+
+
+@Entity(tableName = "News")
+data class CachedData(
+    @PrimaryKey(autoGenerate = true)
+    val id:Int,
```
Comment (RemonShehata) [inline app/src/main/java/com/example/news/model/CachedData.kt]: let this have default value of 0

**Review by RemonShehata — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/news/data/local/Converter.kt [hunk 6]
```diff
@@ -0,0 +1,41 @@
+package com.example.news.data.local
+
+import com.example.news.model.Article
+import com.example.news.model.CachedArticle
+import com.example.news.model.CachedData
+import com.example.news.model.NewsApi
+
+//class Converter {
```
Comment (RemonShehata) [inline app/src/main/java/com/example/news/data/local/Converter.kt]: FYI you still need the converter if you are going to insert a list into room

Comment (RadwaMohamed29) [reply app/src/main/java/com/example/news/data/local/Converter.kt]: done

**Pull request merged**